### DISCUSSION TITLE
Why maxUsableResponseSizeMb is 1.5?

### DISCUSSION THREAD

**Discussion by JC-Ge** [warrenday/graphql-network-inspector#76] [Q&A] [answered]
I'm learning the source code.
When i read config.ts,i find limit max response size to 1.5 MB.
I want to know how to get this number？Are there specific considerations and reasons？

**Comment by warrenday** [comment, marked as answer]
When we parse the response to show it visually as formatted JSON we are converting it to a bunch of html elements. 1.5mb of text turns into about 8mb of html. Beyond this size the app was often crashing/freezing as parsing was so slow. The limit was set to avoid this.

I have offloaded large file payloads to a worker, but even still there were performance issues when rendering that markup on the main thread.

https://github.com/warrenday/graphql-network-inspector/blob/master/src/hooks/useHighlight/index.ts

Possible solutions would be incremental rendering with virtual scrolling. So only render what is on screen.

**Reply by JC-Ge** [reply]
Thank you for your answer
